core/src/history.rs

```diff
@@ -22,18 +25,62 @@ impl ExchangeHistory {
             .map(|(event, batch_id)| (event, BatchId(batch_id as _)))
             .collect();
 
-        Ok(ExchangeHistory { _events: events })
+        Ok(ExchangeHistory::from_events(events))
+    }
+
+    fn from_events(events: Vec<(batch_exchange::Event, BatchId)>) -> Self {
+        // NOTE: Since history methods depend on events being sorted, assert
+        // this invariant is true!
+        debug_assert!(events.windows(2).all(|event_pair| {
```

Comment:
In case someone is wondering about this. There is an `is_sorted` function in std but it doesn't allow specifying a comparison function. That function is called `is_sorted_by` but it is not yet stable.

Reply:
The other option is to use `itertools` which adds this. I wouldn't mind adding the dependency as I can see it being useful in other situations.